config/opal_config_pthreads.m4

```diff
@@ -251,9 +252,10 @@ AC_PROVIDE_IFELSE([AC_PROG_CC],
                   [OPAL_INTL_POSIX_THREADS_PLAIN_C],
                   [opal_pthread_c_success=1])
 
-AC_PROVIDE_IFELSE([AC_PROG_CXX],
-                  [OPAL_INTL_POSIX_THREADS_PLAIN_CXX],
-                  [opal_pthread_cxx_success=1])
+AS_IF([test "$WANT_MPI_CXX_SUPPORT" = 1],
```

Comment:
I see `AC_PROVIDE_IFELSE` used in a few places throughout the tree, but aside from `opal_config_pthreads.m4`, all of them are 3rd-party locations (e.g., libtool, libevent, romio, etc.).

Do you want to get rid of the use of `AC_PROVIDE_IFELSE` here in this file, and instead, use the proper "if we have FOO support" OMPI shell variables?  (e.g., `$WANT_MPI_CXX_SUPPORT`)